dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -21,15 +21,19 @@ import '../framework/utils.dart';
 /// Must match flutter_driver/lib/src/common.dart.
 ///
 /// Redefined here to avoid taking a dependency on flutter_driver.
-String _testOutputDirectory(String testDirectory) {
-  return Platform.environment['FLUTTER_TEST_OUTPUTS_DIR'] ?? '$testDirectory/build';
+String _testOutputDirectory(String testDirectory, bool unittestMode) {
+  if (!unittestMode) {
```

Comment:
What does unittestMode mean? In general I am concerned that this change is increasing the overall cognitive complexity here. It seems like the location of test output directory is the result of two variables, the (essentially global) env var FLUTTER_TEST_OUTPUTS_DIR and the boolean `unittestMode` that we are passing around.

Reply:
unittestmode means that perf_tests fuctions are being called from the end to end test (perf_tests_test.dart) and that it relies on `$testDirectory/build` being the output directory because it tries to parse the resulting metrics file and run a few assertions on its content.

Should I rename unittestMode to testOutputDirectory? we may still need to pass it to the multiple functions in perf_test but at least it will be easier to understand.

Open to suggestions, the solution we decide to use here may need to be applied to some other tests with the same behavior. e.g. tests_test.dart.

Reply:
Yes, I would prefer Goderbauer's suggestion in https://github.com/flutter/flutter/pull/146000#discussion_r1548254343

Reply:
Found a better implementation using the default path everywhere and just copying to the output logs directory if `FLUTTER_TEST_OUTPUTS_DIR` is set.